static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -0,0 +1,434 @@
+    if (useTailor) {
+        return isRegular().then(
+            regular =>
+                regular ? acquisitionsCopyRegulars : acquisitionsCopyControl
+        );
+    }
+
+    return Promise.resolve(acquisitionsCopyControl);
+};
+
+const getCampaignCode = (
+    campaignCodePrefix,
+    campaignID,
+    id,
+    campaignCodeSuffix
+) => {
+    const suffix = campaignCodeSuffix ? `_${campaignCodeSuffix}` : '';
+    return `${campaignCodePrefix}_${campaignID}_${id}${suffix}`;
+};
+
+const addTrackingCodesToUrl = (base: string, campaignCode: string) => {
+    const params = {
+        REFPVID: (config.ophan && config.ophan.pageViewId) || 'not_found',
+        INTCMP: campaignCode,
+    };
+
+    return `${base}?${constructURLQuery(params)}`;
+};
+
+const makeEvent = (id: string, event: string): string => `${id}:${event}`;
+
+const registerIframeListener = (iframeId: string) => {
+    window.addEventListener('message', message => {
+        const iframe = document.getElementById(iframeId);
+
+        if (iframe) {
+            try {
+                const data = JSON.parse(message.data);
+
+                if (data.type === 'set-height' && data.value) {
+                    iframe.style.height = `${data.value}px`;
+                }
+            } catch (e) {
+                // Nothing we can do in the error case
+            }
+        }
+    });
+};
+
+const makeABTestVariant = (
+    options: Object,
+    parentTest: ContributionsABTest
+): Variant => {
+    const trackingCampaignId = parentTest.epic
+        ? `epic_${parentTest.campaignId}`
+        : parentTest.campaignId;
+    const campaignCode = getCampaignCode(
+        parentTest.campaignPrefix,
+        parentTest.campaignId,
+        options.id,
+        parentTest.campaignSuffix
+    );
+    const iframeId = `${parentTest.campaignId}_iframe`;
+
+    const {
+        id,
+
+        // optional params
+        maxViews = defaultMaxViews,
+        isUnlimited = false,
+        contributeURL = addTrackingCodesToUrl(
+            contributionsBaseURL,
+            campaignCode
+        ),
+        membershipURL = addTrackingCodesToUrl(membershipBaseURL, campaignCode),
+        templateFunction = controlTemplate,
+        testimonialBlock = getTestimonialBlock(
+            acquisitionsTestimonialParametersControl
+        ),
+        blockEngagementBanner = false,
+        engagementBannerParams = {},
+        isOutbrainCompliant = false,
+        usesIframe = false,
+        onInsert = noop,
+        onView = noop,
+        useTailoredCopyForRegulars = false,
+        insertAtSelector = '.submeta',
+        insertMultiple = false,
+        insertAfter = false,
+        test = noop,
+        impression = submitImpression =>
+            mediator.once(parentTest.insertEvent, submitImpression),
+        success = submitSuccess =>
+            mediator.once(parentTest.viewEvent, submitSuccess),
+    } = options;
```

Comment:
You could potentially do this up in the function params. I wouldn't worry about these not being `const`s as the linter should catch any attempt to reassign them ([`no-param-reassign`](http://eslint.org/docs/rules/no-param-reassign))